src/applications/vaos/newAppointmentFlow.js

```diff
@@ -149,6 +157,51 @@ export default {
     next: 'vaFacility',
     previous: 'typeOfCare',
   },
+  typeOfEyeCare: {
+    url: '/new-appointment/choose-eye-care',
+    async next(state, dispatch) {
+      const data = getFormData(state);
+      const communityCareEnabled = vaosCommunityCare(state);
+
+      // check that the result does have a ccId
+      if (getTypeOfCare(data)?.ccId !== undefined) {
+        try {
+          if (communityCareEnabled) {
+            // Check if user registered systems support community care...
+            const userSystemIds = await getSystemIdentifiers();
+            const ccSites = await getSitesSupportingVAR(userSystemIds);
+            const ccEnabledSystems = userSystemIds.filter(id =>
+              ccSites.some(site => site.id === id),
+            );
+            dispatch(updateCCEnabledSystems(ccEnabledSystems));
+
+            // Reroute to VA facility page if none of the user's registered systems support community care.
+            if (ccEnabledSystems.length) {
+              const response = await getCommunityCare(getCCEType(state));
+
+              dispatch(updateCCEligibility(response.eligible));
+
+              if (response.eligible) {
+                return 'typeOfFacility';
```

Comment:
I missed this until I ran the branch locally, but you'll need to change the `back` functions for both the `typeOfFacility` and the `vaFacility` page to go back to the `typeOfEyeCare` page when appropriate. You can see this locally by choosing eye care, then picking one of the types, then clicking the back button on the page.